src/pages/degen-swap.tsx

```diff
@@ -0,0 +1,93 @@
+import { Box, Button, Input } from '@chakra-ui/react';
+import React, { useEffect, useState } from 'react';
+import Layout from '../layout';
+import {loadStdlib} from '@reach-sh/stdlib';
+import * as backend from '../contracts/build/degen-swap.main.mjs';
+import { ALGO_WalletConnect as WalletConnect} from '@reach-sh/stdlib';
+import { ALGO_MyAlgoConnect as MyAlgoConnect } from '@reach-sh/stdlib';
+import useWalletAccount from '../utils/persistAccount';
+
+const DegenSwap = () => {
+      const stdlib = loadStdlib('ALGO');
+
+  const [account, _ ] = useWalletAccount();
+  useEffect(() => {
+    console.log(account);
+  }, [account]);
+
+  const [sellerAccount, setSellerAccount] = useState();
+  useEffect(async () => {
+    const stdlib = loadStdlib('ALGO');
+    if (account && account?.address) {
+      if (account?.provider === 'MyAlgo') {
+        stdlib.setWalletFallback(stdlib.walletFallback({
+          providerEnv: 'TestNet', MyAlgoConnect }));
+      } else {
+        stdlib.setWalletFallback(stdlib.walletFallback({
+          providerEnv: 'TestNet', WalletConnect }));
+      }
+      const acct = await stdlib.connectAccount({ addr: account?.address });
+      setSellerAccount(acct);
+    }
+  }, [account]);
+
+  const [nftId, setnftId] = useState();
+  const launchContract = async () => {
+    try {
+      const a = await sellerAccount.contract(backend).p.Seller({
+        getSwap: () => {
+          return [nftId, 89054166, 100, 1_000_000];
+        },
+        cancel: await cancelContract(),
+      });
+    } catch (err) {
+      console.log(err);
+    }
+  }
+
+  const updateNftId = (e) => {
+    setnftId(e.target.value);
+  }
+
+  const [appId, setAppId] = useState(0);
+  const updateApp = (e) => {
+    console.log(e.target.value);
+    setAppId(e.target.value);
+  }
+
+  const connectToApp = async () => {
+    const acct2 = await stdlib.connectAccount({ addr: account?.address });
+
+    const c = await acct2.contract(backend, appId);
+    const connect = await acct2.contract(backend, c.getInfo()).p.Buyer({
+      accSwap: async (t1, t2, p) => { 
+        // returns info
+        console.log(t1,t2,p);
+        // nothing happens
+        return true;
```

Comment:
Good example here, Buyer can acc swap, console does print NFT ID, ASA ID, and Price in ASA but nothing happens at return true but accSwap signature is `accSwap: Fun([Token, Token, UInt], Bool),`

Reply:
Got some advice from Nick to double check:
1. no need to use await on line 61
2. Line 62: also don't think you need the await here, can get the user stuck in the front end. He suggested using the backend.Buyer syntax as well, didn't think that the current syntax would work... backend.Buyer(c, {put interact object here});

Reply:
If you want to use the new syntax, I think it should be: c.p.Buyer(backend, {interact object}). The way it is now you make a copy of the contract handle (e.g. acct2.contract(backend, c.getInfo()))

Reply:
Two other points: 
1. Interact object is missing methods, such as cancel
2. The reach compiler and stdlib have different versions, use version 0.1.10-rc.4